### PR TITLE
MM-30871: Add multi-stage docker build

### PR DESCRIPTION
**Pull request by agnivade**
We add the build stage to ensure consistency in docker images.

https://mattermost.atlassian.net/browse/MM-30871

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,22 @@
...
+RUN CGO_ENABLED=0 make build
+
+#####
+
 FROM alpine
 
-ADD bifrost /bifrost
-ADD config/config.json config/config.json
+COPY --from=builder /go/src/bifrost/bifrost /bifrost
+COPY --from=builder /go/src/bifrost/config/config.json config/config.json
```
Comment (agnivade) [inline Dockerfile:21]: A question that I have is there is no `EXPOSE` directive here. How is the port exposed?

Comment (isacikgoz) [reply Dockerfile:21]: Is it any different from defining when running the container? (i.e. `-p`)

**Review by isacikgoz — APPROVED** [review]
Looks good to me 👍

File: Dockerfile (hunk 1 above)
Comment (agnivade) [reply Dockerfile:21]: That's only to bind the internal port to an external port right? But the expose directive has to be there first. Maybe I am wrong though.

Comment (isacikgoz) [reply Dockerfile:21]: It [seems](https://docs.docker.com/engine/reference/builder/#expose) it has some documentation purposes as well. It doesn't need to be there to being able to bind a port.

Comment (agnivade) [reply Dockerfile:21]: Oh wow, TIL. Thanks @isacikgoz :)

**Review by gabrieljackson — APPROVED** [review]
Awesome!

**Pull request merged**